File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<int> st;
        int count = 0;
        int maxcount = 0;
        for(int i = 0 ; i< s.size() ; i++){
            if(s[i] == '(') st.push(s[i]);
            else if(s[i] == ')') {
                count = st.size();
                st.pop();
                maxcount = max(maxcount , count);
            }
        }
        return maxcount;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        n = 0
        count = 0
        for c in s:
            if c == '(':
                count += 1
            n = max(count , n)    
            if c == ')':
                count -= 1
        return n
                
